### PR TITLE
Fix 404 error when remove self from an organization

### PR DESCRIPTION
**Pull request by yp05327**
Same to #24322 

Not only `leave` action but also `remove` action should check whether user still belongs to the org.

### CODE REVIEW

**Review by wxiaoguang — APPROVED** [review]
Overall LGTM. ps: there is a handy function: `ctx.JSONRedirect(....)`

File: routers/web/org/teams.go [hunk 1]
```diff
@@ -92,12 +92,13 @@ func TeamsAction(ctx *context.Context) {
...
+				redirect = setting.AppSubURL + "/"
+			} else {
+				redirect = ctx.Org.Organization.HomeLink()
+			}
 		}
-		ctx.JSON(http.StatusOK,
-			map[string]any{
-				"redirect": redirect,
-			})
+		ctx.JSON(http.StatusOK, map[string]any{"redirect": redirect})
```
Comment (wxiaoguang) [inline routers/web/org/teams.go]:
```suggestion
		ctx.JSONRedirect(redirect)
```

File: routers/web/org/teams.go [hunk 2]
```diff
@@ -124,10 +125,18 @@ func TeamsAction(ctx *context.Context) {
...
+			ctx.ServerError("IsOrganizationMember", err)
+			return
+		} else if !isOrgMember {
+			if ctx.Org.Organization.Visibility.IsPrivate() {
+				redirect = setting.AppSubURL + "/"
+			} else {
+				redirect = ctx.Org.Organization.HomeLink()
+			}
+		}
+		ctx.JSON(http.StatusOK, map[string]any{"redirect": redirect})
```
Comment (wxiaoguang) [inline routers/web/org/teams.go]:
```suggestion
		ctx.JSONRedirect(redirect)
```

**Review by lunny — APPROVED** [review]

File: routers/web/org/teams.go [hunk 3]
```diff
@@ -124,10 +125,18 @@ func TeamsAction(ctx *context.Context) {
...
+			ctx.ServerError("IsOrganizationMember", err)
+			return
+		} else if !isOrgMember {
+			if ctx.Org.Organization.Visibility.IsPrivate() {
+				redirect = setting.AppSubURL + "/"
+			} else {
+				redirect = ctx.Org.Organization.HomeLink()
+			}
+		}
+		ctx.JSONRedirect(redirect)
```
Comment (delvh) [inline routers/web/org/teams.go]: What about extracting this check into a separate method and giving it a parameter `defaultRedirect`?
You call it twice at the moment (90-101 and 128-139).

Comment (yp05327) [reply routers/web/org/teams.go]: Done.

**Pull request merged**